Code/GraphMol/Canon.cpp

```diff
@@ -1250,6 +1256,796 @@ void canonicalizeFragment(ROMol &mol, int atomIdx,
+    // enhanced possibilties.  They also do not change if the re-entrant call is
+    // mae
+
+    std::string dummyProp;
+    if (!mol->getAtomWithIdx(0)->getPropIfPresent("_canonicalRankingNumber",
+                                                  dummyProp)) {
+      Canon::rankMolAtoms(*mol, ranks, breakTies, includeChirality,
+                          includeIsotopes, includeAtomMaps, useNonStereoRanks,
+                          includeChiralPresence, includeStereoGroups);
+
+      for (auto atom : mol->atoms()) {
+        atom->setProp("_canonicalRankingNumber", ranks[atom->getIdx()]);
+      }
+    }
+
+    OrderedSet<RankedValue> allRankedValues;
+
+    std::vector<StereoGroup> groupsToProcess;
+    std::vector<StereoGroup> andGroupsToKeep;
+
+    for (auto &grp : mol->getStereoGroups()) {
+      if (stereoGroupType == grp.getGroupType()) {
+        groupsToProcess.push_back(grp);
+      } else if (stereoGroupType == StereoGroupType::STEREO_OR &&
+                 grp.getGroupType() == StereoGroupType::STEREO_AND) {
+        andGroupsToKeep.push_back(grp);
+      }
+    }
+
+    mol->setStereoGroups(
+        andGroupsToKeep);  // these groups might be empty, especially if we
+                           // are PROCESSING AND groups
+    std::unique_ptr<RWMol> bestNewMol;
+    auto newMolCount = std::pow(2, groupsToProcess.size());
+
+    for (unsigned int molIndex = 0; molIndex < newMolCount; ++molIndex) {
+      auto newMol = std::unique_ptr<RWMol>(new RWMol(*(mol.get())));
+
+      for (unsigned int grpIndex = 0; grpIndex < groupsToProcess.size();
+           ++grpIndex) {
+        if (molIndex & (1 << grpIndex)) {
+          for (auto atomPtr : groupsToProcess.at(grpIndex).getAtoms()) {
+            if (atomPtr->getChiralTag() == Atom::CHI_TETRAHEDRAL_CW) {
+              newMol->getAtomWithIdx(atomPtr->getIdx())
+                  ->setChiralTag(Atom::CHI_TETRAHEDRAL_CCW);
+            } else if (atomPtr->getChiralTag() == Atom::CHI_TETRAHEDRAL_CCW) {
+              newMol->getAtomWithIdx(atomPtr->getIdx())
+                  ->setChiralTag(Atom::CHI_TETRAHEDRAL_CW);
+            }
+          }
+          // do any  atropisomer bonds in this stereo group
+
+          for (auto bond : groupsToProcess.at(grpIndex).getBonds()) {
+            if (bond->getStereo() == Bond::STEREOATROPCW) {
+              newMol->getBondWithIdx(bond->getIdx())
+                  ->setStereo(Bond::Bond::STEREOATROPCCW);
+            } else if (bond->getStereo() == Bond::STEREOATROPCCW) {
+              newMol->getBondWithIdx(bond->getIdx())
+                  ->setStereo(Bond::Bond::STEREOATROPCW);
+            }
+          }
+        }
+      }
+
+      if (andGroupsToKeep.size() > 0) {
+        canonicalizeStereoGroups_internal(newMol, StereoGroupType::STEREO_AND,
+                                          StereoGroupAbsOptions::NeverInclude);
+      }
+
+      newMol->updatePropertyCache(true);
```

Comment:
Why is this here? It doesn't look like you have done anything to change the valence states of any atoms.

Reply:
Not there anymore.